### PR TITLE
[SYCL] Update test scripts and sycl-ls to use 'fpga' with ONEAPI_DEVICE_SELECTOR

### PR DESCRIPTION
**Pull request by uditagarwal97**
As per the ONEAPI_DEVICE_SELECTOR [documentation](https://github.com/intel/llvm/blob/sycl/sycl/doc/EnvironmentVariables.md#oneapi_device_selector), the device type can only be cpu, gpu, or fpga (or any combination of those). Currently, 'acc' is also accepted by ONEAPI_DEVICE_SELECTOR as a valid device type, which is incorrect.

This PR modifies lit.cfg.py and sycl-ls  to use 'fpga' instead of 'acc' in ONEAPI_DEVICE_SELECTOR.

### CODE REVIEW

**Comment by aelovikov-intel** [comment]
I think `sycl-ls` should be fixed instead of this.

File: sycl/test-e2e/lit.cfg.py [hunk 1]
```diff
@@ -590,7 +590,7 @@
 config.intel_driver_ver = {}
 for sycl_device in config.sycl_devices:
     env = copy.copy(llvm_config.config.environment)
-    env["ONEAPI_DEVICE_SELECTOR"] = sycl_device
+    env["ONEAPI_DEVICE_SELECTOR"] = sycl_device.replace(":acc", ":fpga")
```
Comment (aelovikov-intel) [inline sycl/test-e2e/lit.cfg.py:593]: Line 430 should be updated instead, I think.

**Comment by uditagarwal97** [comment]
> I think `sycl-ls` should be fixed instead of this.

Yup. I've changed 'sycl-ls' ouput.

**Review by aelovikov-intel — APPROVED** [review]

**Comment by uditagarwal97** [comment]
@intel/llvm-gatekeepers the PR is ready!

**Pull request merged**